asmcomp/selectgen.mli

```diff
@@ -49,7 +49,7 @@ module Effect_and_coeffect : sig
   val none : t
   val arbitrary : t
 
-  val effect : t -> Effect.t
+  val effect_ : t -> Effect.t
```

Comment:
since we're not reserving the `effect` keyword any more, this rename is unnecessary in selectgen.ml[i] for this diff